xclim/utils.py

```diff
@@ -124,6 +128,192 @@ def daily_downsampler(da, freq='YS'):
     return buffer.groupby('tags')
 
 
+def get_ev_length(ev, verbose=True, method=2, timing=True, using_dask_loop=True):
```

Comment:
This is throwing a PEP8 error (C901) due to cyclomatic complexity. Too many functions and conditional statements. We can either suspend that check (as we will likely be writing even more complex indices) or we can try refactoring this function to externalize functions that might be called by other indices in the future. Thoughts?

Reply:
This is @sbiner 's stuff ; ) I'll let him handle it.

Reply:
I'm resolving since does not affect the current PR. I think I based my PR out of his branch to avoid conflicts.